vhdbuilder/packer/post-install-dependencies.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+OS=$(sort -r /etc/*-release | gawk 'match($0, /^(ID_LIKE=(coreos)|ID=(.*))$/, a) { print toupper(a[2] a[3]); exit }')
+UBUNTU_OS_NAME="UBUNTU"
+
+source /home/packer/provision_installs.sh
+source /home/packer/provision_installs_distro.sh
+source /home/packer/provision_source.sh
+source /home/packer/provision_source_distro.sh
+source /home/packer/tool_installs.sh
+source /home/packer/tool_installs_distro.sh
+
+CPU_ARCH=$(getCPUArch)  #amd64 or arm64
+VHD_LOGS_FILEPATH=/opt/azure/vhd-install.complete
+
+if [[ $OS == $MARINER_OS_NAME ]]; then
+  if grep -q "kata" <<< "$FEATURE_FLAGS"; then
+    enableMarinerKata
+  else
+    # Leave automatic package update disabled for the kata image
+    enableDNFAutomatic
+  fi
+fi
+
+if [[ $OS == $UBUNTU_OS_NAME ]]; then
+  # shellcheck disable=SC2021
+  current_kernel="$(uname -r | cut -d- -f-2)"
+  dpkg --get-selections | grep -e "linux-\(headers\|modules\|image\)" | grep -v "$current_kernel" | tr -s '[[:space:]]' | tr '\t' ' ' | cut -d' ' -f1 | xargs -I{} apt-get remove -yq {}
```

Comment:
please sanity check the running kernel listed in the final release notes vs the apt packages in the release notes // the logs from these steps

I want to be super careful to make sure we 1) reboot before this and 2) never remove the latest kernel // make sure apt/dpkg/ubuntu fail _hard_ if we do so by accident.